includes/Classifai/Providers/OpenAI/ChatGPT.php

```diff
@@ -0,0 +1,510 @@
+				'max'           => 5,
+				'min'           => 1,
+				'step'          => 1,
+				'default_value' => $default_settings['length'],
+				'description'   => __( 'How many sentences should the excerpt be?', 'classifai' ),
+			]
+		);
+
+		add_settings_field(
+			'temperature',
+			esc_html__( 'Temperature value', 'classifai' ),
+			[ $this, 'render_input' ],
+			$this->get_option_name(),
+			$this->get_option_name(),
+			[
+				'label_for'     => 'temperature',
+				'input_type'    => 'number',
+				'max'           => 2,
+				'min'           => 0,
+				'step'          => 0.1,
+				'default_value' => $default_settings['temperature'],
+				'description'   => __( 'What sampling temperature to use, between 0 and 2. Higher values like 1.8 will make the output more random, while lower values like 0.2 will make it more focused and deterministic.', 'classifai' ),
+			]
+		);
+	}
+
+	/**
+	 * Sanitization for the options being saved.
+	 *
+	 * @param array $settings Array of settings about to be saved.
+	 *
+	 * @return array The sanitized settings to be saved.
+	 */
+	public function sanitize_settings( $settings ) {
+		$new_settings  = $this->get_settings();
+		$authenticated = $this->authenticate_credentials( $settings['api_key'] ?? '' );
+
+		if ( is_wp_error( $authenticated ) ) {
+			add_settings_error(
+				'api_key',
+				'classifai-auth',
+				$authenticated->get_error_message(),
+				'error'
+			);
+
+			$new_settings['authenticated'] = false;
+
+			// For response code 429, credentials are valid but rate limit is reached.
+			if ( 429 === (int) $authenticated->get_error_code() ) {
+				$new_settings['authenticated'] = true;
+			}
+		} else {
+			$new_settings['authenticated'] = true;
+		}
+
+		$new_settings['api_key'] = sanitize_text_field( $settings['api_key'] ?? '' );
+
+		if ( empty( $settings['enable_excerpt'] ) || 1 !== (int) $settings['enable_excerpt'] ) {
+			$new_settings['enable_excerpt'] = 'no';
+		} else {
+			$new_settings['enable_excerpt'] = '1';
+		}
+
+		if ( isset( $settings['length'] ) && is_numeric( $settings['length'] ) && (int) $settings['length'] >= 0 && (int) $settings['length'] <= 5 ) {
+			$new_settings['length'] = absint( $settings['length'] );
+		} else {
+			$new_settings['length'] = 2;
+		}
+
+		if ( isset( $settings['temperature'] ) && is_numeric( $settings['temperature'] ) && (float) $settings['temperature'] >= 0 && (float) $settings['temperature'] <= 2 ) {
+			$new_settings['temperature'] = abs( (float) $settings['temperature'] );
+		} else {
+			$new_settings['temperature'] = 1;
+		}
+
+		return $new_settings;
+	}
+
+	/**
+	 * Authenticate our credentials.
+	 *
+	 * @param string $api_key Api Key.
+	 *
+	 * @return bool|WP_Error
+	 */
+	protected function authenticate_credentials( string $api_key = '' ) {
+		// Check that we have credentials before hitting the API.
+		if ( empty( $api_key ) ) {
+			return new WP_Error( 'auth', esc_html__( 'Please enter your API key', 'classifai' ) );
```

Comment:
Could adjust to "Please enter your OpenAI ChatGPT API key." by including the service provider & feature as well as including the trailing period

Reply:
Updated in [9542a02](https://github.com/10up/classifai/pull/405/commits/9542a02ebf8fd899babb02794e209748dc76903e)